node/impl/full/actor_events.go

```diff
@@ -321,30 +338,30 @@ func (a *ActorEventHandler) SubscribeActorEvents(ctx context.Context, evtFilter
 				case <-ctx.Done():
 					return
 				case <-ticker.C:
-					if params.MaxHeight > 0 && a.Chain.GetHeaviestTipSet().Height() > params.MaxHeight {
+					currentHeight := a.chain.GetHeaviestTipSet().Height()
+					if params.MaxHeight > 0 && currentHeight > params.MaxHeight {
```

Comment:
yep, I think this could be dealt with if we wanted to mess with the `EventFilterManager` but I'm hesitant to get too much more tangled up in there

it would also be helpful to batch by height, as you've suggested, but that would be done in `EventFilterManager` too

Reply:
You're right.